lib/trailblazer/finder/activity/prepare_adapters.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Trailblazer
+  class Finder
+    module Activity
+      # Adapters Activity
+      class PrepareAdapters < Trailblazer::Activity::Railway
+        def check_for_adapters(ctx, **)
+          adapters = ctx.dig(:config, :adapters)
+          return true if adapters.empty?
+
+          adapters.each do |adapter|
+            return true if Finder::Adapters.constants.include?(adapter.to_sym)
+          end
+          false
+        end
+
+        ## Only one ORM can be accepted
+        def validate_adapters(ctx, **)
+          adapters = ctx[:config][:adapters]
+          return true if adapters.empty?
+
+          adapters.each do |adapter|
+            if ORM_ADAPTERS.include?(adapter)
+              return false if (adapters & (ORM_ADAPTERS - [adapter])).any?
+            end
+          end
+        end
+
+        def invalid_adapters_error((ctx, flow_options), **_circuit_options)
+          (ctx[:errors] ||= []) << {adapters: "One or more of the specified adapters are invalid"}
```

Comment:
it could be useful to return which adapters are invalid - what do you think?

Reply:
That will change the api.
The next change i will do is to not merge the paginators with orms.
This error will be irrelevant.